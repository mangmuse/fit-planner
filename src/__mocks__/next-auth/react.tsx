import React from "react";

export const SessionProvider = ({
  children,
}: {
  children: React.ReactNode;
}) => {
  return <>{children}</>;
};

export const useSession = jest.fn(() => ({
  data: {
    user: { id: "user123", email: "test@example.com", image: "/" },
    expires: "2024-12-31T23:59:59.999Z",
  },
  status: "authenticated",
  update: jest.fn(),
}));

export const signIn = jest.fn();
export const signOut = jest.fn();
export const getSession = jest.fn();
