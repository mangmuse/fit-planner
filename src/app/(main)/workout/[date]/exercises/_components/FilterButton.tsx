import { clsx } from "clsx";

export type FilterButtonProps<T extends string> = {
  label: T;
  onClick: (value: T) => void;
  isSelected: boolean;
};

function FilterButton<T extends string>({
  label,
  onClick,
  isSelected,
}: FilterButtonProps<T>) {
  return (
    <button
      role="tab"
      aria-selected={isSelected}
      onClick={() => onClick?.(label)}
      className={clsx(
        "text-xs rounded-md h-8 min-w-[60px] px-3 transition-all",
        isSelected
          ? "bg-primary text-text-black font-medium"
          : "bg-[#212121] text-text-muted"
      )}
    >
      {label}
    </button>
  );
}

export default FilterButton;
