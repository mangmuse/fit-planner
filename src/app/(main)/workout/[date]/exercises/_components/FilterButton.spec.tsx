import { fireEvent, render, screen } from "@testing-library/react";
import FilterButton from "./FilterButton";
const renderFilterButton = ({
  label = "test",
  isSelected,
  onClick,
}: {
  label?: string;
  isSelected?: boolean;
  onClick?: () => void;
} = {}) => {
  return render(
    <FilterButton label={label} isSelected={isSelected} onClick={onClick} />
  );
};
describe("FilterButton", () => {
  it("버튼 클릭시 props로 전달받은 label과 함께 onClick을 호출한다", () => {
    const mockOnClick = jest.fn();
    const label = "test";

    renderFilterButton({ label, onClick: mockOnClick });

    const button = screen.getByText(label);
    fireEvent.click(button);

    expect(mockOnClick).toHaveBeenCalledWith(label);
  });

  it("props로 전달받은 label이 버튼에 렌더링된다", () => {
    const label = "test";

    renderFilterButton({ label });

    expect(screen.getByText(label)).toBeInTheDocument();
  });

  it("props로 전달받은 isSelected가 true일 경우 버튼에 bg-primary, text-text-black 클래스명이 추가된다", () => {
    const label = "test";

    const { container } = renderFilterButton({ label, isSelected: true });

    const button = container.firstChild as HTMLElement;
    expect(button).toHaveClass("bg-primary", "text-text-black");
  });

  it("props로 전달받은 isSelected가 false일 경우 bg-[#212121] 클래스명이 추가된다", () => {
    const label = "test";

    const { container } = renderFilterButton({ label, isSelected: false });

    const button = container.firstChild as HTMLElement;
    expect(button).toHaveClass("bg-[#212121]");
  });

  it("기본 클래스명이 올바르게 적용된다", () => {
    const label = "test";

    const { container } = renderFilterButton({ label });

    const button = container.firstChild as HTMLElement;
    expect(button).toHaveClass(
      "text-[10px]",
      "rounded-md",
      "w-12",
      "min-w-12",
      "max-w-12",
      "h-[28px]",
      "px-1"
    );
  });
});
