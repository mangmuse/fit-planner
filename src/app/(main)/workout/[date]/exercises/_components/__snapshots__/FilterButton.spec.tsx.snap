// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot isSelected가 false일 때 렌더링 결과가 스냅샷과 일치하는지 확인한다 1`] = `
<DocumentFragment>
  <button
    class="text-[10px] rounded-md w-12 min-w-12 max-w-12 h-[28px] px-1 bg-[#212121]"
    data-is-selected="false"
  >
    test
  </button>
</DocumentFragment>
`;

exports[`snapshot isSelected가 true일 때 렌더링 결과가 스냅샷과 일치하는지 확인한다 1`] = `
<DocumentFragment>
  <button
    class="text-[10px] rounded-md w-12 min-w-12 max-w-12 h-[28px] px-1 bg-primary text-text-black"
    data-is-selected="true"
  >
    test
  </button>
</DocumentFragment>
`;
